terraform-provider-constellation/internal/provider/shared_attributes.go

```diff
@@ -166,6 +170,9 @@ func newImageAttributeSchema(t attributeType) schema.Attribute {
 				MarkdownDescription: "Semantic version of the image.",
 				Computed:            !isInput,
 				Required:            isInput,
+				Validators: []validator.String{
```

Comment:
we should also allow complete image references (e.g. ` "ref/main/stream/nightly/v2.14.0-pre.0.20231214193540-2c50abcc919b"`), which are needed for tests too

Reply:
This semver check is superfluous with the other check, which checks if the reference is either a semver, or a versionsapi short path, like the one you've sent. Thanks!